services/galley/src/Galley/API/Teams.hs

```diff
@@ -136,8 +141,9 @@ uncheckedDeleteTeam zusr zcon tid = do
         let e = newEvent TeamDelete tid now
         let r = list1 (userRecipient zusr) (membersToRecipients (Just zusr) membs)
         pushSome ((newPush1 zusr (TeamEvent e) r & pushConn .~ zcon) : events)
+        when ((view teamBinding . Data.tdTeam <$> team) == Just Binding) $
+            deleteUsers (map (view userId) membs)
```

Comment:
If a crash occurs at this point, users will be deleted, but the team itself will not be marked as deleted, meaning a retry of the request isn't possible (since a retry would fail with a 403 due to checking permissions). Perhaps mark team as deleted before deleting users?